wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -355,13 +355,16 @@ public final class MLSService: MLSServiceInterface {
         ) { [weak self] _ in
             guard
                 let self,
-                let context = context,
-                ZMUser.selfUser(in: context).selfClient()?.hasRegisteredMLSClient == true else {
-                self?.logger.info("Skip periodic key material check since MLS is not enabled")
+                let context = context else {
                 return
             }
 
-            Task {
+            Task { [context] in
+                guard await context.perform({ ZMUser.selfUser(in: context).selfClient()?.hasRegisteredMLSClient == true }) == true else {
+                    self.logger.info("Skip periodic key material check since MLS is not enabled")
+                    return
+                }
```

Comment:
Why are you moving the guard condition inside the `Task`?

Reply:
I don't recall exactly 😅. let me check

Reply:
This is called on main and the context is the syncContext, so we need to do perform.

if I do a performAndWait I don't want to block the mainThread.

@typfel  In any case I don't see it to be a blocker, do you?
or you mean it could be a race condition of the check?

Reply:
No, it's not a blocker it's just that we always spawn a Task in this case.